tests/func-tests/node_placement_test.go

```diff
@@ -73,14 +152,16 @@ var _ = Describe("[rfe_id:4356][crit:medium][vendor:cnv-qe@redhat.com][level:sys
 				delete(expectedWorkloadsPods, "secondary-dns")
 			}
 
-			By("Listing pods in infra node")
-			pods := listPodsInNode(client, workloadsNode.Name)
+			Eventually(func(g Gomega) {
+				By("Listing pods in infra node")
+				pods := listPodsInNode(client, workloadsNode.Name)
 
-			By("Collecting nodes of pods")
-			updatePodAssignments(pods, expectedWorkloadsPods, "workload", workloadsNode.Name)
+				By("Collecting nodes of pods")
+				updatePodAssignments(pods, expectedWorkloadsPods, "workload", workloadsNode.Name)
 
-			By("Verifying that all expected workload pods exist in workload nodes")
-			Expect(expectedWorkloadsPods).ToNot(ContainElement(false))
+				By("Verifying that all expected workload pods exist in workload nodes")
+				Expect(expectedWorkloadsPods).ToNot(ContainElement(false))
+			}).WithTimeout(5 * time.Minute).WithPolling(30 * time.Second).Should(Succeed())
```

Comment:
@nunnatsa @tiraboschi do you have a different suggestion for the timeout and poll intervals?